filter/compatibility.go

```diff
@@ -0,0 +1,20 @@
+package filter
+
+type Compatibility struct {
+	RegexTreatment  RegexTreatment
+	SingleStarMatch SingleStarMatch
+}
+
+type RegexTreatment bool
+
+const (
+	StrictStartMatch RegexTreatment = false
+	LooseStartMatch  RegexTreatment = true
+)
+
+type SingleStarMatch bool
+
+const (
+	SingleStarMatchDisabled    SingleStarMatch = false
```

Comment:
maybe private?

and godoc

Reply:
It is used in filter, when feature flags are parsed from string into these enums